core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -733,8 +734,10 @@ protected final boolean performAllBuildSteps(BuildListener listener, Iterable<?
 
         private void reportError(BuildStep bs, Throwable e, BuildListener listener, boolean phase) {
             String msg = "Publisher " + bs.getClass().getName() + " aborted due to exception";
-            e.printStackTrace(listener.error(msg));
-            LOGGER.log(WARNING, msg, e);
+            if (!(e instanceof AbortException)){
```

Comment:
> The equivalent pattern in the past (of returning `false` from the `perform()` method) left no message

Sure, but then the plugin author was _deciding_ to leave no message (or more likely to print one to a `BuildListener` earlier). If they _decided_ to throw `AbortException`, they supplied a message, so we should use it.